src/Lib.hs

```diff
@@ -391,6 +387,11 @@ retirePool dataLayer poolId = convertIOToHandler $ do
     retiredPoolId <- addRetiredPool poolId
 
     return . ApiResult $ retiredPoolId
+
+addPool :: DataLayer -> PoolId -> PoolMetadataHash -> PoolMetadataWrapped -> Handler (ApiResult DBFail PoolId)
+addPool dataLayer poolId poolHash (PoolMetadataWrapped poolMetadataJson) =
+  fmap ApiResult $ convertIOToHandler $
+    (fmap . second) (const poolId) $ runPoolInsertion dataLayer poolMetadataJson poolId poolHash
```

Comment:
Please split it up into several lines to make it more readable. I know you can understand this as it is, but again, it hurts readability.

Reply:
I think newlines hurt readability by adding noise. It also makes scrolling through code much harder. But I'm not the one deciding the style here :)